### PR TITLE
[16024] Add ROS utilities tutorial

### PR DESCRIPTION
**Pull request by dcoello-eProsima**
Signed-off-by: David Coello Pulido <davidcoello@eprosima.com>

### CODE REVIEW

File: docs/rst/tutorials/micro/utilities/utilities.rst [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+===================
+
+.. contents:: Table of Contents
+    :depth: 1
+    :local:
+    :backlinks: none
+
+`micro_ros_utilities <https://github.com/micro-ROS/micro_ros_utilities>`_ it is a package that provides utilities for easing the use of micro-ROS on different platforms. Currently it has two `API <https://micro.ros.org/docs/api/utils/>`_:
+
+- ``micro-ROS`` type initializer based on :ref:`rosidl_typesupport_introspection_c <docs/ros2_documentation/source/Concepts/About-Internal-Interfaces.rst>_`.
```
Comment (dcoello-eProsima) [inline docs/rst/tutorials/micro/utilities/utilities.rst:13]: How can I link this local file?

**Review by pablogs9 — CHANGES_REQUESTED** [review]

File: docs/rst/tutorials/micro/utilities/utilities.rst [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+micro_ros_utilites provides a functionality to instntiate sequences and strings of fixed sizes.
+
+Memory configuration struct:
+
+- max_string_capacity: Maximum string capacity to use for msg fields in case they don’t have a custom rule assigned to them.
+- max_ros2_type_sequence_capacity: Maximum capacity to use for sequence type msg fields (ie: unbounded arrays and lists) which contain ROS 2 msg types, in case they don’t have a custom rule assigned to them.
+- max_basic_type_sequence_capacity: Maximum capacity to use for sequence type msg fields (ie: unbounded arrays and lists) which contain basic types (ie: primitive field types), in case they don’t have a custom rule assigned to them.
+
```
[the remark this answers is not in the document]
Comment (dcoello-eProsima) [reply docs/rst/tutorials/micro/utilities/utilities.rst]: where I can find more info about this

**Comment by pablogs9** [comment]
Depends on https://github.com/eProsima/vulcanexus/pull/93

Please @Acuadros95 review before Friday 25th nov

**Comment by Acuadros95** [comment]
Could we solve this TODOS with this tutorial?:
```rst
`Handling messages memory tutorial <https://micro.ros.org/docs/tutorials/advanced/handling_type_memory>`_

.. TODO(pgarrido): replace this tutorial with the one in vulcanexus when available
.. TODO(acuadros95): Add reference to future "Handling messages memory" tutorial
```

Or will `micro-ROS memory handling` tutorial address this?

**Comment by pablogs9** [comment]
@Acuadros95  yes, we can reference it

**Comment by Acuadros95** [comment]
@pablogs9 After checking the tutorial, this is more like an API explanation. We should probably address that TODOs on the memory tutorial

**Comment by pablogs9** [comment]
@dcoello-eProsima could you address @Acuadros95 comments?

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+
+micro-ROS Types Utilities
+-------------------------
+
+This API helps developers to manage ROS types in micro-ROS. It handles the types structures recursively in order to initialize each member with the required memory size.
+
+.. note::
+
+    A full example can be found on how to use this API can be found `here <https://github.com/micro-ROS/micro_ros_arduino/blob/humble/examples/micro-ros_types_handling/micro-ros_types_handling.ino>`_
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:190]: Same example given in both sections. Move this to the header of the section.

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+.. code-block:: c
+
+    static micro_ros_utilities_memory_conf_t conf = {0};
+
+    // OPTIONALLY this struct can configure the default size of strings, basic sequences and composed sequences
+    conf.max_string_capacity = 50;
+    conf.max_ros2_type_sequence_capacity = 5;
+    conf.max_basic_type_sequence_capacity = 5;
+
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:216]:
```suggestion
- ``max_string_capacity``: Maximum string capacity to use for message fields.
- ``max_ros2_type_sequence_capacity``: Maximum capacity to use for sequence type msg fields (ie: unbounded arrays and lists) which contain ROS 2 msg types.
- ``max_basic_type_sequence_capacity``: Maximum capacity to use for sequence type msg fields (ie: unbounded arrays and lists) which contain basic types (ie: primitive field types).

.. code-block:: c

    static micro_ros_utilities_memory_conf_t conf = {0};
    // OPTIONALLY this struct can configure the default size of strings, basic sequences and composed sequences
    conf.max_string_capacity = 50;
    conf.max_ros2_type_sequence_capacity = 5;
    conf.max_basic_type_sequence_capacity = 5;

All message members will follow this configuration, unless they have a custom rule assigned to them.

```

Moved repeated sentence.

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    #include <control_msgs/msg/joint_jog.h>
+
+    control_msgs__msg__JointJog msg;
+    rosidl_runtime_c__String ros_str = micro_ros_utilities_type_info(ROSIDL_GET_MSG_TYPE_SUPPORT(control_msgs, msg, JointJog));
+
+micro_ros_utilities_get_static_size
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Returns the static memory size that will be used for a type.
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:232]:
```suggestion
Returns the static memory size that will be used for a type with a given memory configuration.
```

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+micro_ros_utilities_destroy_message_memory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Deallocates the dynamic memory of a message.
+
+.. code-block:: c
+
+    #include <control_msgs/msg/joint_jog.h>
+
+    uint8_t my_buffer[1000];
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:306]:
```suggestion
```

Not used in this code snippet

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+micro_ros_utilities_type_info
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Returns a ``rosidl_runtime_c__String`` with the type introspection data.
+
+.. code-block:: c
+
+    #include <control_msgs/msg/joint_jog.h>
+
+    control_msgs__msg__JointJog msg;
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:226]:
```suggestion
```

Not used in this code snippet

File: docs/rst/microros_documentation/user_api/user_api_utilities.rst [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+
+micro_ros_string_utilities_set
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create a ``rosidl_runtime_c__String`` from a ``char`` pointer reallocating an actual ``rosidl_runtime_c__String``.
+
+.. code-block:: c
+
+    const char * str = "Hello World";
+    size_t size = strlen(str);
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_utilities.rst:81]:
```suggestion 
    size_t size = strlen(str) + 1;  // Add null terminator
```

**Pull request merged**